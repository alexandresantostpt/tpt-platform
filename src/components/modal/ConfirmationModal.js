import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import { colors } from '@helpers/colors'
import Icon, { StyledIcon } from '@components/icon/Icon'
import ClickOutside from './ClickOutside'
import { i18n } from '@i18n'

const Button = styled.button`
    background-color: ${colors.white};
    border: solid 0.8px ${colors.lightBlueGray};
    border-radius: 1.5rem;
    color: ${colors.lightBlueGray};
    height: 26.3px;
    margin: 1rem;
    outline: none;
    padding: 0.1875rem 2.5rem;
    text-transform: uppercase;
    width: 112.3px;
    &:hover {
        background-color: ${colors.white};
        border: solid 0.8px ${colors.purple};
        color: ${colors.purple};
    }
`
const ButtonContainer = styled.div`
    flex-direction: row;
    margin-top: 1rem;
`

const CenterIcon = styled.div`
    display: flex;
    justify-content: center;
    width: 100%;
`
const CloseButton = styled.button`
    background-color: transparent;
    border: none;
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    outline: none;
    width: 27rem;
    ${StyledIcon} {
        align-self: center;
        color: ${colors.link};
        margin: 0.2rem;
        &:hover {
            color: ${colors.purple};
        }
    }
`
const Content = styled.div`
    align-items: center;
    background-color: ${colors.white};
    border-radius: 8px;
    box-shadow: 4px 4px 10px 0 rgba(0, 0, 0, 0.18);
    display: flex;
    flex-direction: column;
    justify-content: center;
    max-height: 180px;
    width: 27rem;
`
const IconLayout = styled.div`
    background: linear-gradient(-225deg, #5271c4 0%, #b19fff 48%, #eca1fe 100%);
    border-radius: 30rem;
    display: flex;
    height: 21px;
    justify-content: center;
    width: 21px;
    ${StyledIcon} {
        align-self: center;
        color: ${colors.white};
        margin-right: 0.1px;
        &:hover {
            color: ${colors.white};
        }
    }
`

const ModalContainer = styled.div`
    align-items: center;
    background-color: rgba(0, 0, 0, 0.1);
    display: flex;
    height: 100%;
    justify-content: center;
    left: 0;
    overflow: auto;
    position: fixed;
    top: 0;
    width: 100%;
    z-index: 1;
`

const ConfirmationModal = ({ openModal, confirmationAction, close, item, icon, children }) => (
    <>
        {openModal ? (
            <ModalContainer>
                <ClickOutside onClick={() => close()}>
                    <Content>
                        <CloseButton>
                            <Icon fontSize="1.5rem" icon="times" onClick={() => close()} pointer={true} />
                        </CloseButton>
                        <CenterIcon>
                            <IconLayout>
                                <Icon fontSize="1.2rem" icon={icon} pointer={true} />
                            </IconLayout>
                        </CenterIcon>
                        {children}
                        <ButtonContainer>
                            <Button
                                onClick={function() {
                                    confirmationAction(item)
                                    close()
                                }}
                            >
                                {i18n.t('buttons.yes')}
                            </Button>
                            <Button onClick={() => close()}> {i18n.t('buttons.no')}</Button>
                        </ButtonContainer>
                    </Content>
                </ClickOutside>
            </ModalContainer>
        ) : null}
    </>
)
ConfirmationModal.defaultProps = {
    close: null,
    item: null
}

ConfirmationModal.propTypes = {
    children: PropTypes.array.isRequired,
    close: PropTypes.func,
    confirmationAction: PropTypes.func.isRequired,
    icon: PropTypes.string.isRequired,
    item: PropTypes.object,
    openModal: PropTypes.bool.isRequired
}

export default ConfirmationModal
