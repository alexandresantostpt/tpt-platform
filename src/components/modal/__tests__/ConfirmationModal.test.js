import React from 'react'
import renderer from 'react-test-renderer'

import ConfirmationModal from '../ConfirmationModal'

describe('Tests for ConfirmationModal component', () => {
    it('Should render without crash', () => {
        const tree = renderer.create(<ConfirmationModal openModal={true} />).toJSON()
        expect(tree).toMatchSnapshot()
    })
})
