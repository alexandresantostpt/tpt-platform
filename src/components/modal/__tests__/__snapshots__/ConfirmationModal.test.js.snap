// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for ConfirmationModal component Should render without crash 1`] = `
<div
  className="ConfirmationModal__ModalContainer-sc-1rdkds7-6 hwtbW"
>
  <div
    className="ConfirmationModal__Content-sc-1rdkds7-4 eBCQjc"
  >
    <button
      className="ConfirmationModal__CloseButton-sc-1rdkds7-3 dwEcRc"
    >
      <span
        className="icon icon-times Icon__StyledIcon-crr1cg-0 dqcjNO"
        disabled={false}
        fontSize="1.5rem"
        onClick={[Function]}
      />
    </button>
    <div
      className="ConfirmationModal__CenterIcon-sc-1rdkds7-2 ggEkby"
    >
      <div
        className="ConfirmationModal__IconLayout-sc-1rdkds7-5 hUsVys"
      >
        <span
          className="icon icon-undefined Icon__StyledIcon-crr1cg-0 htHnBz"
          disabled={false}
          fontSize="1.2rem"
          onClick={null}
        />
      </div>
    </div>
    <div
      className="ConfirmationModal__ButtonContainer-sc-1rdkds7-1 gmmNDJ"
    >
      <button
        className="ConfirmationModal__Button-sc-1rdkds7-0 mupnl"
        onClick={[Function]}
      >
        Sim
      </button>
      <button
        className="ConfirmationModal__Button-sc-1rdkds7-0 mupnl"
        onClick={[Function]}
      >
         
        NÃ£o
      </button>
    </div>
  </div>
</div>
`;
